modules/auxiliary/admin/mysql/mysql_enum.rb

```diff
@@ -53,6 +55,7 @@ def report_cred(opts)
 
   def run
     return if not mysql_login_datastore
```

Comment:
doesn't `mysql_login_datastore` attempt to login to the mysql instance? seems like we'd want to skip that step entirely if we have a session

Oh, I've just taken a look at the changes to `mysql_login` over here https://github.com/rapid7/metasploit-framework/pull/18718/files#diff-2aa11fddd9e25ca4117395fe4855d555acecfdc39f60ab4efc5d751e81f5c336R35-R40
I'm not sure I'd expect a call to `mysql_login` _not_ to attempt a log in, feels to me more like you'd want to not attempt the call to `mysql_login` if you already know you have a session, what do you think?

Reply:
Currently addressing this after our sync up meeting. Will be pushed up as part of the next commit.